src/App.php

```diff
@@ -1151,4 +1182,101 @@ private function outputResponseJson($data): void
         $this->setResponseHeader('Content-Type', 'application/json');
         $this->outputResponse($data);
     }
+
+    /**
+     * Return true if $_GET[$key] exists.
+     */
+    public function hasRequestGetParam(string $key): bool
```

Comment:
It seems you completely missed docs, they cannot be checked using CI, so for every change you commit, you must keep them in sync!

Reply:
you are right, unit test failing makes me do some too fast cut/paste.
I update them